temperature/crud.py

```diff
@@ -0,0 +1,65 @@
+from sqlalchemy.orm import Session
+
+from temperature import models, schemas
+
+
+def create_temperature(
+    db: Session, temperature: schemas.TemperatureCreateUpdate
+) -> models.Temperature:
+    db_temperature = models.Temperature(**temperature.model_dump())
+    db.add(db_temperature)
+    db.commit()
+    db.refresh(db_temperature)
+
+    return db_temperature
+
+
+def get_temperature(db: Session, temperature_id: int) -> models.Temperature:
+    return (
+        db.query(models.Temperature)
+        .filter(models.Temperature.id == temperature_id).first()
+    )
+
+
+def get_temperature_list(
+    db: Session, city_id: int = None, skip: int = 0, limit: int = 10
+) -> list[models.Temperature]:
+    query = db.query(models.Temperature)
+
+    if city_id:
+        query = query.filter(models.Temperature.city_id == city_id)
+
+    return query.offset(skip).limit(limit).all()
+
+
+def update_temperature(
+    db: Session,
+    temperature_id: int,
+    temperature: schemas.TemperatureCreateUpdate
+) -> models.Temperature:
+    db_temperature = (
+        db.query(models.Temperature)
+        .filter(models.Temperature.id == temperature_id).first()
+    )
+
+    if db_temperature:
```

Comment:
In the `update_temperature` function, ensure that the `temperature_id` exists before attempting to update. Although the current code checks if `db_temperature` is not None, it might be useful to raise an exception or return a specific response if the temperature record is not found.